### ISSUE TITLE
Error starting EMR cluster using AWS CLI

### ISSUE THREAD

**Issue by SrikanthTati** [aws/aws-cli#2552]
I'm trying to add AWS command to my SBT build script to create a Spark cluster. The command fails with error "_Error parsing parameter '--instance-groups': Expected: '=', received: ''' for input:_"

Sample command I tried

```
aws emr create-cluster \
  --debug \
  --name Pixel-chopup-kinesis-0.1 --auto-scaling-role EMR_AutoScaling_DefaultRole --applications Name=Hadoop Name=Spark Name=Ganglia \
  --service-role EMR_DefaultRole --enable-debugging --release-label emr-5.4.0 --log-uri 's3n://my-bucket/path' \
  --scale-down-behavior TERMINATE_AT_INSTANCE_HOUR --region us-east-1 \
  --instance-groups '[{"InstanceCount":3,"InstanceGroupType":"CORE","InstanceType":"m3.xlarge","Name":"Core - 2"}]'

```

> CLI version: aws-cli/1.11.36 Python/2.7.12 Linux/4.4.0-59-generic botocore/1.4.93

I've enabled --debug and see this

> 2017-04-16 11:11:46,624 - MainThread - awscli.argprocess - DEBUG - Parsing param --instance-groups as shorthand
> 2017-04-16 11:11:46,625 - MainThread - awscli.clidriver - DEBUG - Exception caught in main()
> Traceback (most recent call last):
>   File "/usr/local/lib/python2.7/dist-packages/awscli/clidriver.py", line 197, in main
>     return command_table[parsed_args.command](remaining, parsed_args)
>   File "/usr/local/lib/python2.7/dist-packages/awscli/clidriver.py", line 333, in __call__
>     return command_table[parsed_args.operation](remaining, parsed_globals)
>   File "/usr/local/lib/python2.7/dist-packages/awscli/customizations/commands.py", line 166, in __call__
>     cli_argument=cli_argument, value=value, operation=None)
>   File "/usr/local/lib/python2.7/dist-packages/botocore/session.py", line 691, in emit_first_non_none_response
>     responses = self._events.emit(event_name, **kwargs)
>   File "/usr/local/lib/python2.7/dist-packages/botocore/hooks.py", line 227, in emit
>     return self._emit(event_name, kwargs)
>   File "/usr/local/lib/python2.7/dist-packages/botocore/hooks.py", line 210, in _emit
>     response = handler(**kwargs)
>   File "/usr/local/lib/python2.7/dist-packages/awscli/argprocess.py", line 317, in __call__
>     cli_argument, value, service_name, operation_name)
>   File "/usr/local/lib/python2.7/dist-packages/awscli/argprocess.py", line 341, in _parse_as_shorthand
>     raise ParamError(cli_argument.cli_name, str(e))
> ParamError: Error parsing parameter '--instance-groups': Expected: '=', received: ''' for input:
> '[{"InstanceCount":3,"InstanceGroupType":"CORE","InstanceType":"m3.xlarge","Name":"Core
> ^
> 2017-04-16 11:11:46,625 - MainThread - awscli.clidriver - DEBUG - Exiting with rc 255
> 
> Error parsing parameter '--instance-groups': Expected: '=', received: ''' for input:
> '[{"InstanceCount":3,"InstanceGroupType":"CORE","InstanceType":"m3.xlarge","Name":"Core

Full log can be seen [here](https://www.dropbox.com/s/fom2j4c0nyp2z9h/AWS_CLI_ERROR.txt?dl=0)

**Comment by JordonPhillips** [comment]
Is that exactly what you used? I had to add in the `\` after each line to make it work. I ask because I see that in the logs it says you passed in:

```
'\'[{"InstanceCount":3,"InstanceGroupType":"CORE","InstanceType":"m3.xlarge","Name":"Core', '-', '2"}]\''
```

Which would be interpreted as a string, and so wouldn't work.

**Comment by SrikanthTati** [comment]
I'm was not running this from CLI but from inside SBT script. Its like launching a shell command from inside a script. Those  extra """ '\' """ that you see are not there in the command string I passed. I thought AWS CLI added it automatically. I did read some other issues with shorthand where that was recommended. Since I hit this, I've moved on to use API instead of CLI in my script. I'm no longer blocked on this. Feel free to close this if you don't see any issues or if you think its incorrect usage .
